scripts/mainmenu.js

```diff
@@ -1,115 +1,147 @@
+		$.RegisterForUnhandledEvent('ChaosShowPauseMenu', this.onShowPauseMenu.bind(this));
+		$.RegisterForUnhandledEvent('ChaosHidePauseMenu', this.onHidePauseMenu.bind(this));
+		$.RegisterForUnhandledEvent('MapSelector_OnLoaded', this.onMapSelectorLoaded.bind(this));
+		$.RegisterForUnhandledEvent('Safeguard_Disconnect', this.onSafeguardDisconnect.bind(this));
+		$.RegisterForUnhandledEvent('Safeguard_Quit', this.onSafeguardQuit.bind(this));
+		$.RegisterForUnhandledEvent('Safeguard_ChangeMap', this.onSafeguardMapChange.bind(this));
+		$.RegisterForUnhandledEvent('ReloadBackground', this.setMainMenuBackground.bind(this));
+		$.RegisterForUnhandledEvent('OnMomentumQuitPrompt', this.onQuitPrompt.bind(this));
+		$.RegisterEventHandler('Cancelled', $.GetContextPanel(), this.onEscapeKeyPressed.bind(this));
 
 		// Close the map selector when a map is successfully loaded
 		$.RegisterForUnhandledEvent(
 			'MapSelector_TryPlayMap_Outcome',
-			(outcome) => outcome && MainMenuController.onHomeButtonPressed()
+			(outcome) => outcome && this.onHomeButtonPressed()
 		);
 
 		$.DispatchEvent('ChaosHideIntroMovie');
 	}
 
-	// TODO: Delete me when proper engine support is added!
-	static psdump() {
-		for (let i = 0; i < $.persistentStorage.length; i++) {
-			$.Msg(
-				$.persistentStorage.key(i).toString() +
-					': ' +
-					JSON.stringify($.persistentStorage.getItem($.persistentStorage.key(i)))
-			);
-		}
-	}
+	/**
+	 * General onLoad initialisations.
+	 * Fired when ChaosMainMenu fires it's onload event.
+	 */
+	static onMainMenuLoaded() {
+		// These aren't accessible until the page has loaded fully, find them now
+		this.panels.movie = $('#MainMenuMovie');
+		this.panels.model = $('#MainMenuModel');
+
+		this.panels.model.SetModelRotation(0.0, 270.0, 0.0); // Get arrow logo facing to the right, looks better
+		this.panels.model.SetModelRotationSpeedTarget(0.0, 0.2, 0.0);
+		this.panels.model.SetMouseXRotationScale(0.0, 1.0, 0.0); // By default mouse X will rotate the X axis, but we want it to spin Y axis
+		this.panels.model.SetMouseYRotationScale(0.0, 0.0, 0.0); // Disable mouse Y movement rotations
 
-	static onInitFadeUp() {
-		if (!MainMenuController.playedInitialFadeUp) {
-			MainMenuController.playedInitialFadeUp = true;
+		this.panels.model.LookAtModel();
+		this.panels.model.SetCameraOffset(-200.0, 0.0, 0.0);
+		this.panels.model.SetCameraFOV(40.0);
+
+		this.panels.model.SetDirectionalLightColor(0, 0.5, 0.5, 0.5);
+		this.panels.model.SetDirectionalLightDirection(0, 1.0, 0.0, 0.0);
+
+		if (GameInterfaceAPI.GetSettingBool('developer')) {
+			$('#ControlsLibraryButton').RemoveClass('hide');
+			$('#PSDumpButton').RemoveClass('hide');
 		}
+
+		this.setMainMenuBackground();
+
+		this.showPlaytestWelcomePopup();
 	}
 
+	/**
+	 * Fired by C++ whenever main menu is switched to.
+	 */
 	static onShowMainMenu() {
-		MainMenuController.onInitFadeUp();
+		this.panels.movie = $('#MainMenuMovie');
+		this.panels.image = $('#MainMenuBackground');
```

Comment:
Why's this happening here again? Static initialization sets this already.

Reply:
I don't understand why, but loading into a map unsets these, resulting in a black background after you disconnect back to main menu. Easiest fix is to just set them again and call `setMainMenuBackground` again.